### ISSUE TITLE
Movie Label Confusion

### ISSUE THREAD

**Issue by roshankern** [WayScience/mitocheck_data#16]
We noticed another minor issue in the labeled data sent by J-K Heriche (see https://github.com/WayScience/mitocheck_data/issues/2):

The morphological class names in [`trainingset.dat`](https://github.com/WayScience/mitocheck_data/blob/main/0.download_data/training_set/trainingset.dat) differ from the names specified in the [original Mitocheck paper](https://www.nature.com/articles/nature08869#Fig1). 

The table below documents the classes specified in the original paper vs what we assume are the corresponding classes from  `trainingset.dat`:
| Paper         | `trainingset.dat` |
| :-------------: | :-------------:|
| Interphase | Interphase |
| Elongated | Elongated |
| Large | Large |
| Prometaphase | Prometaphase |
| MAP | MetaphaseAlignment |
| Metaphase | Metaphase |
| Anaphase | ??? |
| Binuclear | ??? |
| Polylobed | ??? |
| Grape | Grape |
| Hole | Hole |
| Small Irr | SmallIrregular |
| Folded | Folded |
| Cell Death | Apoptosis |
| Condensed | UndefinedCondensed |
| Artefact | Artefact |

`trainingset.dat` also includes the following 3 morphological classes that we are unsure how to pair with the paper classes:

Shape1, Shape3,  ADCCM

**Comment by roshankern** [comment]
J-K kindly informed us that Shape1 and Shape3 correspond to binuclear and polylobed respectively. This was confirmed by visual analysis (see [example frame](https://github.com/WayScience/mitocheck_data/blob/main/3.extract_features/plateLT0017_19_well365_frame64.png)). 
Furthermore, ADCCM is a class grouping the other classes (ex A: artefact, D: dynamic/folded, C: condensed, etc).

**Comment by roshankern** [comment]
There are differences between the feature dataset sent by J-K and the manual annotations dataset described in ["automatic identification of chromosome morphologies"](https://www.sciencedirect.com/science/article/pii/S1047847709002731?via%3Dihub#aep-section-id21) that are causing confusion.

- Most notably, the `Anaphase` class does not exist in the J-K-sent data. This raises the question: do manual annotations for this class exist as indicated by the [automatic identification of chromosome phenotypes](https://www.sciencedirect.com/science/article/pii/S1047847709002731?via%3Dihub) paper?
- The `ADCCM` class exists in the J-K-sent data but not in the [automatic identification of chromosome phenotypes](https://www.sciencedirect.com/science/article/pii/S1047847709002731?via%3Dihub) paper. What is the point of this class? How should it be classified into one of the 16 classes mentioned in the paper?
- The instances of each label in the the J-K sent data differs from the instances mentioned in the [automatic identification of chromosome phenotypes](https://www.sciencedirect.com/science/article/pii/S1047847709002731?via%3Dihub) paper. In total about 9% of the labels are missing (ignoring the differences in ADCCM/anaphase). What is the cause of these differences? The instances of each label in the J-K-sent data are shown below:

| Label         | Instances |
| :-------------: | :-------------:|
| Interphase | 355 |
| Elongated | 34 |
| Large | 47 |
| Prometaphase | 363 |
| MAP | 196 |
| Metaphase | 75 |
| ADCCM | 105 |
| Binuclear | 291 |
| Polylobed | 478 |
| Grape | 97 |
| Hole | 128 |
| Small Irr | 182 |
| Folded | 51 |
| Cell Death | 170 |
| Condensed | 31 |
| Artefact | 86 |
| Total | 2689 |

The label instances as mentioned by the paper are shown in the confusion matrix for the model, with the total instances of the labels being 2957:

![matrix](https://ars.els-cdn.com/content/image/1-s2.0-S1047847709002731-gr3.jpg)

**Comment by roshankern** [comment]
J-K kindly informed us that the discrepancies in our `trainingset.dat` file and the [automatic identification of chromosome phenotypes](https://www.sciencedirect.com/science/article/pii/S1047847709002731?via%3Dihub) paper are the result of differing versions of the training annotations. The `trainingset.dat` file is an earlier version without the `Anaphase` class.

J-K has also clarified that the `ADCCM` class stands for "Abnormal Distribution of Condensed Chromosome Masses". This class was only observed for one gene and was not included in the training set.

J-K sent us a new file, `traingset_2007_06_21.dat`, that follows the same data format as `trainingset.dat` but has classes matching those in the paper including the `Anaphase` class. The instances of each label in the `traingset_2007_06_21.dat` file are shown below: 

| Label         | Instances |
| :-------------: | :-------------:|
| Interphase | 420 |
| Anaphase | 54 |
| Elongated | 32 |
| Large | 52 |
| Prometaphase | 365 |
| MAP | 207 |
| Metaphase | 47 |
| ADCCM | 100 |
| Binuclear | 302 |
| Polylobed | 473 |
| Grape | 104 |
| Hole | 101 |
| Small Irr | 116 |
| Folded | 34 |
| Cell Death | 183 |
| Condensed | 31 |
| Artefact | 91 |
| Total | 2712 |

Notably, `traingset_2007_06_21.dat` must not be the same training set used in the [automatic identification of chromosome phenotypes](https://www.sciencedirect.com/science/article/pii/S1047847709002731?via%3Dihub) paper as the total number of labels differ by about 250. J-K has mentioned that he could inquire about if a more extended set of annotations is available as some were made after the end of the project.

**Comment by roshankern** [comment]
The following images from plate `LT0013_42`, well `107`, frame `39` demonstrate why there are likely different versions of the `features` dataset.

Frame labeled with data from `trainingset.dat` (older version):
![image](https://user-images.githubusercontent.com/105665675/176002323-a0a7b7a1-0c28-4240-b6c8-05b8893f0299.png)

Frame labeled with data from `trainingset_2007_06_21.dat` (newer version):
![image](https://user-images.githubusercontent.com/105665675/176003187-412d8372-cf71-4dbc-a49b-ee5d083e56ed.png)

The labels from `trainingset.dat`(figure 1) do not seem to correspond well with the actual cell stages (as judged manually). Only with the newer labels from `trainingset_2007_06_21.dat` do the labels seem to correspond well. This indicates that the `features` dataset we have does not correspond to all `trainingset.dat` files, but rather to a particular version of `trainingset.dat`.

If there are different versions of the `features` dataset, it is very important we use the one that aligns with the version of `trainingset.dat` that we use. Thus, it is worth asking Heriche if there are different versions of the `features` dataset and, if so, see if he can confirm that we use matching versions of the `features` dataset and the `trainingset.dat` file. 

We are still having issues aligning the `features` dataset with the newer `trainingset_2007_06_21.dat` file. This is demonstrated by the labeled image from plate `LT0109_38`, well `349`, frame `25` below. Note label `68: Interphase` on the far right:
![image](https://user-images.githubusercontent.com/105665675/176005435-444c5227-0910-44ad-b265-fdab98309299.png)

**Comment by roshankern** [comment]
Dr. Thomas Walter from Mitocheck confirmed that these missing bounding boxes are unexplainable errors.
